lighthouse-core/report/html/renderer/element-screenshot-renderer.js

```diff
@@ -0,0 +1,256 @@
+/**
+ * @license Copyright 2020 The Lighthouse Authors. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ */
+'use strict';
+
+/* globals self Util */
+
+/** @typedef {import('./dom.js')} DOM */
+/** @typedef {LH.Artifacts.Rect} Rect */
+/** @typedef {{width: number, height: number}} Size */
+
+/**
+ * @param {number} value
+ * @param {number} min
+ * @param {number} max
+ */
+function clamp(value, min, max) {
+  if (value < min) return min;
+  if (value > max) return max;
+  return value;
+}
+
+/**
+ * @param {Rect} rect
+ */
+function getRectCenterPoint(rect) {
+  return {
+    x: rect.left + rect.width / 2,
+    y: rect.top + rect.height / 2,
+  };
+}
+
+class ElementScreenshotRenderer {
+  /**
+   * Given the location of an element and the sizes of the preview and screenshot,
+   * compute the absolute positions (in screenshot coordinate scale) of the screenshot content
+   * and the highlighted rect around the element.
+   * @param {Rect} elementRectInScreenshotCoords
+   * @param {Size} elementPreviewSizeInScreenshotCoords
+   * @param {Size} screenshotSize
+   */
+  static getScreenshotPositions(elementRectInScreenshotCoords,
```

Comment:
I would like to push back a little on the idea that anything anywhere here is *clearly* working in any particular scale :)

it's definitely clear*er* what's happening within this function but `getScreenshotPositions` could easily be a name for a function that *returns* `somethingScreenshotCords` based on `somethingDisplayCoords`

are you on board for heavily commenting at least the params and intermediates? keeping the variable name seems like an easy alternative to heavy commenting

Reply:
aiight well if you're already feelin this, lemme pitch you on what I was going to put in a separate PR. basically abbreviating all the suffixes:

* `elementRectInScreenshotCoords` => `elementRect_SC`
* `maxRenderSizeInDisplayCoords` => `maxRenderSize_DC`

On my first pass looking at this renderer code i was really confused because the var names suffix was "Coords" but their types were Rect/Size. But then realized the different coords spaces and i'm pleased the names indicate this signal.  But still, it just takes up so much visual space that I think it current hampers readability.

So yeah. suffix abbreviation. Connor's on board. so if you're into it, we could just do thi.

Reply:
this plus the fileoverview explaining the two coordinate spaces (and what the abbreviation means) SGTM 👍

Reply:
Just no underscores, please :)

Reply:
> So yeah. suffix abbreviation. Connor's on board. so if you're into it, we could just do this.

plz do this in followup pr